test/mocks/compression/compressor/mocks.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "envoy/compression/compressor/compressor.h"
+#include "envoy/compression/compressor/config.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Compression {
+namespace Compressor {
+
+class MockCompressor : public Compressor {
+public:
+  MockCompressor();
+  ~MockCompressor() override;
+
+  // Compressor::Compressor
+  MOCK_METHOD(void, compress, (Buffer::Instance & buffer, State state));
+};
+
+class MockCompressorFactory : public CompressorFactory {
```

Comment:
I guess this is not currently used because the filter test still uses the old Config class which took a factory name rather than the factory pointer?

Reply:
Not sure which Config class you're referring, but I've added a test case dedicated to `HttpFilters::Compressor::CompressorFilterConfig`. There this mock is used to check if a factory's methods are actually called.

Reply:
Ah here, I see it now https://github.com/envoyproxy/envoy/pull/10553/files#diff-a905be3f5872395dc9391e1e82022f87R17